bin_loc[] bin_array